### PR TITLE
Make Transport Property Names more strict (and case-insenstitive)

### LINKED ISSUE #1338: Property names
From the review by Lars Eggert, COMMENT part:

### Section 4.1, paragraph 1
```
    Transport Properties are referred to by property names.  For the
    purposes of this document, these names are alphanumeric strings in
    which the following characters are allowed: lowercase letters a-z,
    uppercase letters A-Z, digits 0-9, the hyphen -, and the underscore
    _. These names serve two purposes:
```
This allows property names that are indistinguishable from integers.
Is that a bug or a feature? Also, are they case-sensitive?

### PR DESCRIPTION
**Pull request by philsbln**
alternative to #1402
closes #1338

### CODE REVIEW

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -563,9 +563,11 @@ the respective protocol has been selected.
 
 ## Transport Property Names {#property-names}
 
-Transport Properties are referred to by property names. These names are alphanumeric strings in which the following
-characters are allowed: lowercase letters `a-z`, uppercase letters `A-Z`,
-digits `0-9`, the hyphen `-`, and the underscore `_`. These names serve two purposes:
+Transport Properties are referred to by property names. These names are
+case-insensitive alphanumeric strings in which the following characters are allowed:
+letters `a-z` / `A-Z`, digits `0-9`, the hyphen `-`.
+The underscore `_` is only allowed at the first position.
```
Comment (mwelzl) [inline draft-ietf-taps-interface.md:569]:
```suggestion
```

I don't understand the point of this rule, and it even contradicts what you say below about snake_case. The next sentence, "They must start with a letter or an underscore" seems good enough.

File: draft-ietf-taps-interface.md [hunk 2]
```diff
@@ -595,6 +597,8 @@ for Protocol-specific Properties and MUST NOT be used for vendor or implementati
 Terms listed as keywords as in the protocol numbers registry SHOULD be avoided as any part of a vendor- or
 implementation-specific property name.
 
+Though Transport Property Names are case-insensitive, it is recommended to use camelCase to improve readability.
+Implementations may transpose Transport Property Names into snake_case or PascalCase to blend into the language environment.
```
Comment (mwelzl) [inline draft-ietf-taps-interface.md:601]: I note that this is pretty strict... we don't know how styles of future languages will be?  But I like it nevertheless. If we allow "any" kind of transposition, we make porting applications harder.

**Review by mwelzl — CHANGES_REQUESTED** [review]
One suggestion

File: draft-ietf-taps-interface.md (hunk 1 above)
Comment (mwelzl) [reply draft-ietf-taps-interface.md:569]: Aha, I see...
```suggestion
The underscore `_` is only allowed at the first position (with the exception of transposing into snake_case, see below).
```

**Review by mwelzl — APPROVED** [review]

File: draft-ietf-taps-interface.md [hunk 3]
```diff
@@ -563,9 +563,11 @@ the respective protocol has been selected.
 
 ## Transport Property Names {#property-names}
 
-Transport Properties are referred to by property names. These names are alphanumeric strings in which the following
-characters are allowed: lowercase letters `a-z`, uppercase letters `A-Z`,
-digits `0-9`, the hyphen `-`, and the underscore `_`. These names serve two purposes:
+Transport Properties are referred to by property names. These names are
+case-insensitive alphanumeric strings in which the following characters are allowed:
+letters `a-z` / `A-Z`, digits `0-9`, the hyphen `-`.
```
Comment (mwelzl) [inline draft-ietf-taps-interface.md:568]:
```suggestion
letters `a-z` / `A-Z`, digits `0-9`, the hyphen `-` and the underscore `_`.
```

**Review by tfpauly — CHANGES_REQUESTED** [review]

**Review by gorryfair — APPROVED** [review]
I like this line of approach - subject to addressing current review comments

**Comment by mwelzl** [comment]
I think this minimalistic and much less strict angle reflects what was agreed on at the interim. About time to merge it, I believe.

**Pull request merged**